clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/deploy/ops/CreateServerGroupAtomicOperation.java

```diff
@@ -334,18 +334,42 @@ protected RegisterTaskDefinitionRequest makeTaskDefinitionRequest(
     return request;
   }
 
-  protected RegisterTaskDefinitionRequest makeTaskDefinitionRequestFromArtifact(
-      String ecsServiceRole, EcsServerGroupName newServerGroupName) {
+  private RegisterTaskDefinitionRequest getSpelProcessedArtifact() {
+    RegisterTaskDefinitionRequest requestTemplate = null;
+    for (Map<Object, Object> serializedSpelProcessedTaskDefArtifact :
```

Comment:
I'm confused why `serializedSpelProcessedTaskDefArtifact` is a list - i'm not sure it needs to be, but if it does, then as written this func will loop through and convert each item in the collection but only return the last one. If we're doing this for future proofing reasons wouldn't we want to retain the results of each conversion?

Reply:
So I have changed the type of spelProcessedTaskDefArtifact from List of Maps to Map of Objects which solves this issue. The code has been updated.